### PR TITLE
d-saber fix / blocking stuff

### PR DESCRIPTION
**Pull request by Tarmunora**
## About the PR 
~~Adds a bit of disorient resist and ranged protection to active c-sabers~~
Also fixes d-sabers to not do upwards of 150 stam damage on every hit

Wow this PR is completely different now! A whole bunch of blocking-related stuff in the backend, mostly in moving a bunch of it over to signal-component system, as well as a couple example (unused) components for custom itemblocks. 
Also cleans up the tooltips for items with special blocks and whatnot

## Why's this needed? 
~~C-saber currently is filling a very similar role to a number of other traitor melee weapons; this aims to give it a little more identity (everybody knows glowing swords protect against bullets, right?)~~

~~This may not be the right direction to give it (arguably the c-saber should be more about the concealability as opposed to rampage), but vov~~

Contains badly needed and overcommented sample code for signal component stuff.
debloats tooltips slightly

### CODE REVIEW

**Comment by PrimeNumb** [comment]
I am not necessarily opposed to this change, but I have some thoughts and questions.
Currently the csaber has the sole role of being a pure murder weapon, which is the role it's
had for as long as I can remember. I think this is fine; some traitor items need to fit into this
category after all.

The katana and red chainsaw also fit into this role (as you mentioned). 
The difference being that:
#### Katana 
Focused on disabling/crippling opponents, 5 successful hits to kill an opponent outright (assuming correct limb targetting).
#### Red chainsaw 
Focused on disabling/crippling opponents as well. Difference from katana is that it mainly targets organs instead. Is much worse at _immediately_ killing opponents than both the csaber and katana.

My question is, where do you think the csaber should fit in? 
You state that the csaber should "be more about the concealability as opposed to rampage", but your changes will absolutely pull it in the rampage direction. I think this is fine too, but it should be noted by the coders approving this that this is the design direction it is taking.

Also, what is the gameplay context that this is meant to address? Can you describe a specific scenario that you're targetting, how it plays out currently and how it would play out post-change? 
Is this an issue, currently?

**Comment by Tarmunora** [comment]
This change was intended as, well, a stopgap measure to make the c-saber not strictly inferior to the katana in what it's used for. If I were to do a sweeping rework of it, it would probably be towards making it more like the NTSO-baton - you pull it out and get couple very powerful swings, but then have to wait for it to regenerate charge or something. Not really sure yet,

**Comment by Tarmunora** [comment]
If @mybluecorners 's blocking PR goes through, this should be folded into that

File: code/datums/components/saberblock.dm [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+			S.setProperty("rangedprot", 1)
+			S.setProperty("disorient_resist", 35)
+datum/component/itemblock/saberblock/on_block_end(datum/source, mob/user)
+	. = ..()
+	var/obj/item/sword/S = src.parent
+	S.delProperty("rangedprot")
+	S.delProperty("disorient_resist")
+
+
+
```
Comment (ZeWaka) [inline code/datums/components/saberblock.dm:18]:
```suggestion
	S.delProperty("disorient_resist")

```

File Cleanup

File: code/datums/components/_itemblock.dm [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+		return COMPONENT_INCOMPATIBLE
+	RegisterSignal(parent, COMSIG_ITEM_BLOCK_BEGIN, .proc/on_block_begin)
+	RegisterSignal(parent, COMSIG_ITEM_BLOCK_END, .proc/on_block_end)
+	var/obj/item/I = src.parent
+	if(istype(I) && bonus)
+		I.setProperty("block_bonus")
+
+
+
+/datum/component/itemblock/proc/on_block_begin(datum/source, mob/user)
```
Comment (ZeWaka) [inline code/datums/components/_itemblock.dm:19]:
```suggestion
		I.setProperty("block_bonus")

/datum/component/itemblock/proc/on_block_begin(datum/source, mob/user)
```
File Cleanup

**Comment by Tarmunora** [comment]
C-saber buff commented out, this PR is more about the underlying blocking stuff.
Doing anything with it can come in another PR

**Comment by ZeWaka** [comment]
A good fix and cleanliness PR. Previous commentary no longer applies to this iteration.

File: code/datums/components/itemblock/warmsup.dm [hunk 3]
```diff
@@ -0,0 +1,40 @@
+//example component for itemblocks - heats up the user on life tick while blocking
+/datum/component/itemblock/warmsup
```
Comment (ZeWaka) [inline code/datums/components/itemblock/warmsup.dm:2]: I feel like this is a misnomer. It doesn't just warm you up, it also gives cold protection.

**Pull request merged**